### PR TITLE
Core/Misc: Custom exceptions will use string_view for constructor and noexcept for what

### PR DESCRIPTION
**Pull request by BertiRean**
**Changes proposed:**

-  Change constructor param of custom exceptions to std::string_view instead std::string or const char*
-  
-  

**Issues addressed:**

Closes #  (insert issue tracker number)

**Tests performed:**
Build Pass
(Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

File: src/common/DataStores/DB2FileLoader.h [hunk 1]
```diff
@@ -178,7 +178,7 @@ struct DB2RecordCopy
 class TC_COMMON_API DB2FileLoadException : public std::exception
 {
 public:
-    DB2FileLoadException(std::string msg) : _msg(std::move(msg)) { }
+    DB2FileLoadException(std::string_view msg) : _msg(msg) { }
```
Comment (Shauren) [inline src/common/DataStores/DB2FileLoader.h:181]: This is a bad change. Don't just change things for the sake of changing and check WHY they are written the way they are.

Almost all of DB2FileLoadException are constructed using Trinity::StringFormat which means that old code directly moves the result of formatting into the exception object (1 instance of the string is ever created with content) while your "improved" version causes unneccessary copies

**Review by Shauren — CHANGES_REQUESTED** [review]

File: src/common/DataStores/DB2FileLoader.h (hunk 1 above)
Comment (BertiRean) [reply src/common/DataStores/DB2FileLoader.h:181]: Thanks by the comment. And for the other exceptions ?

Comment (BertiRean) [reply src/common/DataStores/DB2FileLoader.h:181]: I understand that format return a normal std::string, but when you call the constructor with that param is not a new copy created ?
inline std::string StringFormat(FormatString<Args...> fmt, Args&&... args)
It's not a temporal string
I mean not for the constructor
StringFormat -> Create std::string A, A is passed as param to Constructor of Exception, the constructor create a new str B, and this  B is moved to _str member in class

Comment (Shauren) [reply src/common/DataStores/DB2FileLoader.h:181]: Except that the argument for DB2FileLoadException is constructed using `string&&` constructor using return value of StringFormat (which moves, not copies)

**Pull request merged**